src/include/kernel/time/timer.h

```diff
@@ -53,6 +64,8 @@ struct sys_timer {
 	void *param;
 	unsigned int flags;
 	uint32_t state; /**< do we use timer_set or timer_init_start? */
+
+	struct sys_timer_sharing *timer_sharing;
```

Comment:
Is this only for SMP?
if yes maybe use SMP macro?

Reply:
I think maybe it's useful for NOSMP case and just a pointer. so I did't put this file into SMP.  But happy to make changes

Reply:
Ok, thanks!